zerver/lib/cache.py

```diff
@@ -409,6 +409,7 @@ def flush_realm(sender: Any, **kwargs: Any) -> None:
     realm = kwargs['instance']
     users = realm.get_active_users()
     delete_user_profile_caches(users)
+    cache_delete(realm_last_visible_message_id_cache_key(realm))
```

Comment:
This is only actually necessary if we change the value of `message_visibility_limit`.

Reply:
You can say `if kwargs.get('update_fields') is None or 'message_visibility_limit' in kwargs['update_fields']:` to avoid churning the cache too much.  (See `flush_user_profile` for a similar example..)